sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -745,8 +761,10 @@ public PDone apply(PCollection<T> input) {
       public void populateDisplayData(DisplayData.Builder builder) {
         super.populateDisplayData(builder);
 
+        String prefixString = filenamePrefix.isAccessible()
+            ? filenamePrefix.get() : filenamePrefix.toString();
         builder
-            .addIfNotNull(DisplayData.item("filePrefix", filenamePrefix)
+            .addIfNotNull(DisplayData.item("filePrefix", prefixString)
```

Comment:
We should decide what convention we want to establish around display data for `ValueProvider` parameters. This PR establishes one-- try to register the value, and use the `.toString()` if it's not accessible. If that's the pattern we want to adopt, I agree that it makes sense to bake it into the `DisplayData.Builder` API.

Another option would be to have runners late-bind runtime data into registered display data for ValueProviders. This would involve some new API support as well.

 Or, establish that display data is strictly for construction-time metadata and late-bound values should not be included (we're still be able to capture them as PipelineOptions display data, so this wouldn't be a huge loss)

I believe @ajamato had done some thinking here. Alex, anything to add?